public/app/plugins/datasource/grafana-azure-monitor-datasource/variables.ts

```diff
@@ -91,7 +91,10 @@ export class VariableSupport extends CustomVariableSupport<DataSource, AzureMoni
             }
           default:
             request.targets[0] = queryObj;
-            return lastValueFrom(this.datasource.query(request));
+            const queryResp = await lastValueFrom(this.datasource.query(request));
+            return {
+              data: queryResp.data,
+            };
```

Comment:
Because the `lastValueFrom` was not awaited on, the error was not caught

Reply:
nit: you can just do `return await lastValueFrom(this.datasource.query(request));` isn't it?

Reply:
Unfortunately not 🤷‍♀️  , I tried but the error is still uncaught